### DISCUSSION TITLE
statistics/data science workflow (esp. with plots)?

### DISCUSSION THREAD

**Discussion by ym-han** [helix-editor/helix#745] [Q&A]
Hi, have any of you used helix to do statistics or data science? And if so, how do you usually handle plotting --- do you use a tmux-type thing + some sort of terminal graphics package or something?

**Comment by archseer** [comment]
I don't personally do data science, but you can use all sorts of ascii plotting: https://twitter.com/thingskatedid/status/1316074032379248640

**Comment by smason** [comment]
Would also be interested to know what other people do!

I mostly use Python for these tasks.  I do most exploratory work in Jupyter and only involve Helix when I've got something nailed down enough to turn into a module (i.e. by refactoring the code out of the `.ipynb` JSON file into a real `.py` file).  I'll still pull those module into a Jupyter notebook (often with `autoreload` enabled) to make sure the plots are doing what I expect.

Would be nice to use Helix more, as the Jupyter editor isn't great.  I've hacked [Jupyter's CSS](https://stackoverflow.com/a/34742362/1358308) to make the editor a bit nicer, but not much else.

**Comment by AceofSpades5757** [comment]
Jupyter notebook or lab. I used to use some other tools from time to time, but they were almost always more trouble than they were worth.

**Reply by smason** [reply]
I just use a local notebook server, and only occasionally use the markdown features

having a REPL is mainly so that I don't have to (re)load large datasets every time I make a change to the code, making exploratory work much faster.

the main features I use are its ability to edit, evaluate and persistently maintain the output (whether text or images of plots) of blocks of code.

Jupyter's "block" based interaction is a popular abstraction, but something else might work better though I'm not sure what it would look like
